compiler/rustc_trait_selection/src/traits/structural_match.rs

```diff
@@ -170,6 +170,10 @@ impl<'a, 'tcx> TypeVisitor<'tcx> for Search<'a, 'tcx> {
                 let kind = NonStructuralMatchTyKind::Generator;
                 return ControlFlow::Break(NonStructuralMatchTy { ty, kind });
             }
+            ty::TyAlias(..) => {
+                // Type aliases are transparent and should never be seen by structural match checking.
+                bug!("unexpected TyAlias in TypeVisitor::visit_ty");
```

Comment:
Is this saying `TypeVisitor` makes them transparent? Or that there's a normalization step somewhere?